### PR TITLE
Task-49005_50395: Add the possibility to edit targets from the edit scheduling drawer

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
Prior to this change, redactors and space mangers had the new post drawer with stepper. After this change, the new post drawer is available only for Publisher of web contributors group or administrators with the possibility to edit targets from edit scheduling drawer.

### CODE REVIEW

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 1]
```diff
@@ -83,35 +83,39 @@
...
+  private final static String      PLATFORM_ADMINISTRATORS_GROUP       = "/platform/administrators";
 
-  public static final String       NEWS_DRAFT_VISIBILITY_MIXIN_TYPE              = "mix:draftVisibility";
+  public static final String       NEWS_DRAFT_VISIBILITY_MIXIN_TYPE    = "mix:draftVisibility";
 
-  public static final String       NEWS_DRAFT_VISIBILE_MIXIN_PROP         = "exo:draftVisible";
+  public static final String       NEWS_DRAFT_VISIBILE_MIXIN_PROP      = "exo:draftVisible";
 
-  public static final String       MIX_NEWS_MODIFIERS              = "mix:newsModifiers";
+  public static final String       NEWS_ACTIVITY_VISIBILITY_MIXIN_TYPE = "mix:activityVisibility";
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]:
```suggestion
  public static final String       NEWS_ACTIVITY_VISIBILITY_MIXIN_TYPE = "mix:newsPosting";
```

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 2]
```diff
@@ -83,35 +83,39 @@
...
 
-  public static final String       MIX_NEWS_MODIFIERS_PROP         = "exo:newsModifiersIds";
+  public static final String       NEWS_ACTIVITY_HIDDEN_MIXIN_PROP     = "exo:isHiddenActivity";
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]:
```suggestion
  public static final String       NEWS_ACTIVITY_HIDDEN_MIXIN_PROP     = "exo:newsPosted";
```

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 3]
```diff
@@ -727,6 +731,11 @@ public News convertNodeToNews(Node node, boolean editMode) throws Exception {
     } else {
       news.setDraftVisible(false);
     }
+    if (originalNode.hasProperty(NEWS_ACTIVITY_HIDDEN_MIXIN_PROP)) {
+      news.setHiddenActivity(Boolean.valueOf(node.getProperty(NEWS_ACTIVITY_HIDDEN_MIXIN_PROP).getString()));
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]:
```suggestion
      news.setPosted(Boolean.valueOf(node.getProperty(NEWS_ACTIVITY_HIDDEN_MIXIN_PROP).getString()));
```

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 4]
```diff
@@ -727,6 +731,11 @@ public News convertNodeToNews(Node node, boolean editMode) throws Exception {
...
+    } else {
+      news.setHiddenActivity(false);
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]:
```suggestion
      news.setPosted(false);
```

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 5]
```diff
@@ -1206,6 +1215,18 @@ public boolean canPublishNews() {
             currentIdentity.isMemberOf(PLATFORM_WEB_CONTRIBUTORS_GROUP, PUBLISHER_MEMBERSHIP_NAME);
   }
 
+  /**
+   * Return a boolean that indicates if the current user can post/unpost the news
+   * activity on the stream
+   *
+   * @return if the news activity can be posted or not on the stream
+   */
+  public boolean canChooseTargets(String authenticatedUser) {
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: this should be the same as canPublishNews, why adding that ?

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 6]
```diff
@@ -835,6 +835,26 @@ public Response canPublishNews(@Context HttpServletRequest request) {
...
   }
 
+  @GET
+  @Path("canChooseTargets")
+  @Produces(MediaType.TEXT_PLAIN)
+  @RolesAllowed("users")
+  @ApiOperation(value = "check if the current user can choose a target to display section", httpMethod = "GET", response = Response.class, notes = "This checks if the current user can select a target to display section", consumes = "application/json")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "User ability to choose a target is returned"),
+          @ApiResponse(code = 401, message = "User not authorized to choose a target")})
+  public Response canChooseTargets(@Context HttpServletRequest request) {
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: this should be the same as canPublishNews, why adding that ?

File: services/src/main/resources/locale/portlet/news/News_en.properties [hunk 7]
```diff
@@ -58,6 +58,10 @@ news.composer.stepper.continue=Continue
 news.composer.stepper.back=Back
 news.composer.publishSection.option=Publish in a dedicated section instead
 news.composer.stepper.postedOrPublish.description=Select when the article will be posted and/or published
+news.composer.stepper.chooseTarget.others=others
+news.composer.stepper.chooseTarget.mandatory=* The targets selector is mandatory
```
Comment (azayati) [inline services/src/main/resources/locale/portlet/news/News_en.properties]:
```suggestion
news.composer.stepper.chooseTarget.mandatory=The targets selector is mandatory
```
add the * directly in vue file

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 8]
```diff
@@ -83,35 +83,39 @@
...
+  private final static String      PLATFORM_ADMINISTRATORS_GROUP       = "/platform/administrators";
 
-  public static final String       NEWS_DRAFT_VISIBILITY_MIXIN_TYPE              = "mix:draftVisibility";
+  public static final String       NEWS_DRAFT_VISIBILITY_MIXIN_TYPE    = "mix:draftVisibility";
 
-  public static final String       NEWS_DRAFT_VISIBILE_MIXIN_PROP         = "exo:draftVisible";
+  public static final String       NEWS_DRAFT_VISIBILE_MIXIN_PROP      = "exo:draftVisible";
 
-  public static final String       MIX_NEWS_MODIFIERS              = "mix:newsModifiers";
+  public static final String       NEWS_ACTIVITY_VISIBILITY_MIXIN_TYPE = "mix:newsActivityPosting";
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]:
```suggestion
  public static final String       NEWS_ACTIVITY_POSTING_MIXIN_TYPE = "mix:newsActivityPosting";
```

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 9]
```diff
@@ -83,35 +83,39 @@
...
 
-  public static final String       MIX_NEWS_MODIFIERS_PROP         = "exo:newsModifiersIds";
+  public static final String       NEWS_ACTIVITY_HIDDEN_MIXIN_PROP     = "exo:newsActivityPosted";
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]:
```suggestion
  public static final String       NEWS_ACTIVITY_POSTED_MIXIN_PROP     = "exo:newsActivityPosted";
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 10]
```diff
@@ -176,6 +213,70 @@
             </div>
           </v-stepper-content>
         </v-stepper>
+        <div v-else>
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue:216]: this part should be for simple redactors or managers in order to juste post/don't post and schedule

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 11]
```diff
@@ -1220,6 +1229,11 @@ public News scheduleNews(News news) throws Exception {
       if (scheduledNewsNode == null) {
         throw new ItemNotFoundException("Unable to find a node with an UUID equal to: " + news.getId());
       }
+      if (scheduledNewsNode.canAddMixin(NEWS_ACTIVITY_VISIBILITY_MIXIN_TYPE)
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: should we store this information also when posting immediately ?

**Review by azayati — CHANGES_REQUESTED** [review]
see review

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue (hunk 10 above)
Comment (hassenelkamel-exo) [reply webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue:216]: yes

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java (hunk 11 above)
Comment (hassenelkamel-exo) [reply services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: yes, we need it when we use the edit publishing drawer

File: services/src/main/resources/locale/portlet/news/News_en.properties (hunk 7 above)
Comment (hassenelkamel-exo) [reply services/src/main/resources/locale/portlet/news/News_en.properties]: IMO, If we put it in vue composant, anyone can think that is unintentionally made.

Comment (azayati) [reply services/src/main/resources/locale/portlet/news/News_en.properties]: no I don't think so, the asterisk should not be part of translation

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 12]
```diff
@@ -176,6 +213,70 @@
...
+          <div v-if="canPublishNews" class="d-flex flex-column mt-4 ms-3">
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]: this part should be deleted

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Review by azayati — DISMISSED** [review]
LGTM after QA validation

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**